src/nodes.coffee

```diff
@@ -799,6 +799,12 @@ exports.Block = class Block extends Base
     return nodes[0] if nodes.length is 1 and nodes[0] instanceof Block
     new Block nodes
 
+  ast: (o, level) ->
+    if (level? and level isnt LEVEL_TOP) and @expressions.length
+      return (new Sequence(@expressions).withLocationDataFrom @).ast o, level
```

Comment:
We represent things like `a; b` as a `Block`

But Babel-style ASTs distinguish these (eg the equivalent `a, b` in JS) as `SequenceExpression`s

So when generating the AST for a `Block`, generate a `SequenceExpression` AST node (using the new `Sequence` node class) if the current `level` isnt `LEVEL_TOP`